src/main/scala/com/ing/wbaa/gargoyle/sts/oauth/KeycloakTokenVerifier.scala

```diff
@@ -0,0 +1,46 @@
+package com.ing.wbaa.gargoyle.sts.oauth
+
+import com.ing.wbaa.gargoyle.sts.config.GargoyleKeycloakSettings
+import org.keycloak.RSATokenVerifier
+import org.keycloak.adapters.KeycloakDeploymentBuilder
+import org.keycloak.representations.adapters.config.AdapterConfig
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+trait KeycloakTokenVerifier {
+
+  protected[this] def keycloakSettings: GargoyleKeycloakSettings
+  implicit def executionContext: ExecutionContext
+
+  import scala.collection.JavaConverters._
+
+  def verifyToken(token: BearerToken): Future[VerifiedToken] = {
+    Future {
+      val keycloakToken = RSATokenVerifier.verifyToken(
+        token.value,
+        keycloakDeployment.getPublicKeyLocator.getPublicKey(keycloakSettings.realmPublicKeyId, keycloakDeployment),
+        keycloakDeployment.getRealmInfoUrl
+      )
+
+      VerifiedToken(
```

Comment:
Wouldn't you want to return a Option[VerifiedToken] here for instance? It seems now that you always get back a VerifiedToken, whether the token could be verified or not?

Reply:
Then I'd suggest to:
- catch the exception thrown by the keycloak function here
- log the error thrown
- return an option

That way we separate the concerns a bit better I think. Now it feels like this function always succeeds (java way of working with exceptions) and that obfuscates the check a bit.
If you don't agree we can of course ask Adam for the majority vote;)